src/main/scala/java/security/SecureRandom.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * scalajs-fake-insecure-java-securerandom (https://github.com/scala-js/scala-js-fake-insecure-java-securerandom)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.security
+
+import scala.scalajs.js
+import scala.scalajs.js.typedarray._
+
+// The seed in java.util.Random will be unused, so set to 0L instead of having to generate one
+class SecureRandom() extends java.util.Random(0L) {
+  // Make sure to resolve the appropriate function no later than the first instantiation
+  private val getRandomValuesFun = SecureRandom.getRandomValuesFun
+
+  // setSeed has no effect
+  override def setSeed(x: Long): Unit = ()
+
+  override def nextBytes(bytes: Array[Byte]): Unit = {
```

Comment:
Ugh... What is happening here...

Looking at the documentation of `Random`, you're not allowed to do this: The documentation of `next` states:

> Subclasses should override this, as this is used by all other methods.

Since `SecureRandom` is not final, a subclass may override `next` and expect `nextBytes` to use its overridden implementation.

However, SecureRandom provides its own `nextBytes` method...

Reply:
`next` is final in `SecureRandom`. 😉